src/Lykke.AzureStorage/Tables/Entity/ValueTypesMerging/Strategies/IValueTypeMergingStrategy.cs

```diff
@@ -0,0 +1,20 @@
+﻿using Lykke.AzureStorage.Tables.Entity.PropertyAccess;
+using Microsoft.WindowsAzure.Storage.Table;
+
+namespace Lykke.AzureStorage.Tables.Entity.ValueTypesMerging.Strategies
+{
+    /// <summary>
+    /// Implementation should be thread safe
+    /// </summary>
+    internal interface IValueTypeMergingStrategy
+    {
+        void MarkValueTypePropertyAsDirty(string propertyName);
+        void NotifyEntityWasRead();
+        void NotifyEntityWasWritten();
+        EntityProperty GetEntityProperty(
+            AzureTableEntity entity, 
```

Comment:
An interface can be used ITableEntity instead of the class

Reply:
Nope. `ITableEntity` is external common interface. All infrastructure involved by the `GetEntityProperty` call is intended work exactly with the `AzureTableEntity`. But thanks to this remark I made some code improvements.